### PR TITLE
Fix stack overflow

### PR DESCRIPTION
**Pull request by Aras14HD**
**!Depends on #189, merge after that!**
Fixes the stack overflow in `execute` by removing the recursion of that function.

### CODE REVIEW

File: src/reddit_comment.rs [hunk 1]
```diff
@@ -1887,6 +1731,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore = "reason"]
```
Comment (tolik518) [inline src/reddit_comment.rs:1734]: No need to ignore the test in that PR as overflow is fixed and that test runs in a reasonable time (43-46ms on my machine)

Comment (Aras14HD) [reply src/reddit_comment.rs:1734]: Oh, yeah, forgot to enable them again, ignored them for debugging purposes (took too long with them enabled)

**Comment by Aras14HD** [comment]
Tested to the extremes on my machine, the text was >3MB large but only took around 1s, however the result was incorrect due to integer overflow. After some calculations, I noticed, that this would already happen with "only" 131k characters. Since the depth field is already in an align 8 structure, using a `u32` instead of a `u16` makes no difference (both in size and performance). With that, it is safe for up to 8.5GB input text, nothing that anyone regardless of platform would use.

**Comment by tolik518** [comment]
Merged and deployed
